Xamarin.PropertyEditing.Mac/Controls/Custom/TextFieldSmallButtonContainer.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class SmallButton : NSButton
+	{
+		private NSView previousKeyView;
+		public override NSView PreviousKeyView => this.previousKeyView ?? base.PreviousKeyView;
+
+		internal void SetPreviousKeyView (NSView view)
+		{
+			this.previousKeyView = view;
+		}
+
+		public SmallButton ()
+		{
+			BezelStyle = NSBezelStyle.RegularSquare;
+			ControlSize = NSControlSize.Small;
+			Bordered = false;
+			TranslatesAutoresizingMaskIntoConstraints = false;
+			Cell.HighlightsBy = (int)NSCellStyleMask.ContentsCell;
+		}
+	}
+
+	internal class TextFieldSmallButtonContainer : NSTextField
+	{
+		public List<SmallButton> Buttons { get; } = new List<SmallButton> ();
```

Comment:
There doesn't seem to be any reason for this to be public?

Reply:
Fixed